From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: musa <musa@local>
Date: Wed, 3 Jun 2026 11:47:35 +0000
Subject: [PATCH] MUSA:
 vllm.model_executor.layers.fused_moe.experts.deep_gemm_moe

---
 .../layers/fused_moe/experts/deep_gemm_moe.py          | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/vllm/model_executor/layers/fused_moe/experts/deep_gemm_moe.py b/vllm/model_executor/layers/fused_moe/experts/deep_gemm_moe.py
index 211364d24b..f0179e4a0d 100644
--- a/vllm/model_executor/layers/fused_moe/experts/deep_gemm_moe.py
+++ b/vllm/model_executor/layers/fused_moe/experts/deep_gemm_moe.py
@@ -355,9 +355,12 @@ class DeepGemmExperts(mk.FusedMoEExpertsModular):
         # under cudagraph replay.
         with mk_alignment_scope(align_used):
             mm1_out = _resize_cache(workspace2, (M_sum, N))
+            # MUSA mate ragged_moe_gemm_8bit asserts scale tensors are contiguous.
+            a1q_scale = a1q_scale.contiguous()
+            w1_scale = self.w1_scale.contiguous()
             m_grouped_fp8_gemm_nt_contiguous(
                 (a1q, a1q_scale),
-                (w1, self.w1_scale),
+                (w1, w1_scale),
                 mm1_out,
                 expert_ids,
                 **gemm_kwargs,
@@ -372,9 +375,12 @@ class DeepGemmExperts(mk.FusedMoEExpertsModular):
             )
 
             mm2_out = _resize_cache(workspace2, (M_sum, K))
+            # MUSA mate ragged_moe_gemm_8bit asserts scale tensors are contiguous.
+            a2q_scale = a2q_scale.contiguous()
+            w2_scale = self.w2_scale.contiguous()
             m_grouped_fp8_gemm_nt_contiguous(
                 (a2q, a2q_scale),
-                (w2, self.w2_scale),
+                (w2, w2_scale),
                 mm2_out,
                 expert_ids,
                 **gemm_kwargs,
